hyperspy/misc/date_time_tools.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2016 The HyperSpy developers
+#
+# This file is part of  HyperSpy.
+#
+#  HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+import datetime
+from dateutil import tz, parser
+
+
+def metadata_to_datetime(metadata):
+    date = metadata['General']['date']
+    time = metadata['General']['time']
+    if 'time_zone' in metadata['General']:
+        return parser.parse('%sT%s %s' % (date, time, metadata['General']['time_zone']))
```

Comment:
Note that the ISO standard does not include a space between time and time zone info: https://www.w3.org/TR/NOTE-datetime. However, as long as the parser is happy it shouldn't matter too much.